app/src/util/permission-errors.ts

```diff
@@ -0,0 +1,210 @@
+
+/**
+ * Permission error messages for user-friendly display
+ * Organized by resource type for easier maintenance
+ */
+export const PERMISSION_ERROR_MESSAGES: Record<PermissionErrorCode, string> = {
+  // General permission errors
+  [PERMISSION_ERRORS.INSUFFICIENT_PERMISSIONS]: 'You do not have sufficient permissions to perform this action',
+  [PERMISSION_ERRORS.NO_ORGANIZATION_CONTEXT]: 'Unable to determine organization context for permission check',
+  [PERMISSION_ERRORS.ORGANIZATION_NOT_FOUND]: 'Organization not found',
+  [PERMISSION_ERRORS.ORGANIZATION_INACTIVE]: 'Organization is not active',
+  [PERMISSION_ERRORS.NO_ACCESS_TO_RESOURCE]: 'You do not have access to this resource',
+  [PERMISSION_ERRORS.RESOURCE_NOT_FOUND]: 'The requested resource was not found',
+  
+  // Organization-level errors
+  [PERMISSION_ERRORS.CANNOT_ACCESS_ORGANIZATION]: 'You do not have access to this organization',
+  [PERMISSION_ERRORS.CANNOT_MANAGE_ORGANIZATION]: 'You do not have permission to manage this organization',
+  [PERMISSION_ERRORS.NOT_ORGANIZATION_ADMIN]: 'Organization administrator access required',
+  
+  // Project-level errors
+  [PERMISSION_ERRORS.CANNOT_ACCESS_PROJECT]: 'You do not have access to this project',
+  [PERMISSION_ERRORS.CANNOT_CREATE_CITY]: 'You do not have permission to create cities in this project',
+  [PERMISSION_ERRORS.CANNOT_MANAGE_PROJECT]: 'You do not have permission to manage this project',
+  [PERMISSION_ERRORS.NOT_PROJECT_ADMIN]: 'Project administrator access required',
+  
+  // City-level errors
+  [PERMISSION_ERRORS.CANNOT_ACCESS_CITY]: 'You do not have access to this city',
+  [PERMISSION_ERRORS.CANNOT_CREATE_INVENTORY]: 'You do not have permission to create inventories in this city',
+  [PERMISSION_ERRORS.CANNOT_DELETE_CITY]: 'You do not have permission to delete this city',
+  [PERMISSION_ERRORS.NOT_CITY_COLLABORATOR]: 'City collaborator access required',
+  
+  // Inventory-level errors
+  [PERMISSION_ERRORS.CANNOT_ACCESS_INVENTORY]: 'You do not have access to this inventory',
+  [PERMISSION_ERRORS.CANNOT_EDIT_INVENTORY]: 'You do not have permission to edit this inventory',
+  [PERMISSION_ERRORS.CANNOT_DELETE_INVENTORY]: 'You do not have permission to delete this inventory',
+  
+  // User management errors
+  [PERMISSION_ERRORS.CANNOT_MANAGE_USERS]: 'You do not have permission to manage users',
+  [PERMISSION_ERRORS.CANNOT_INVITE_USERS]: 'You do not have permission to invite users',
+  
+  // Role-specific errors
+  [PERMISSION_ERRORS.COLLABORATOR_ACCESS_RESTRICTED]: 'Collaborator access is limited to assigned cities only',
+  [PERMISSION_ERRORS.PROJECT_ADMIN_SCOPE_LIMITED]: 'Project admin access is limited to assigned projects',
+};
+
+/**
+ * Create a standardized permission error with code and message
+ * Uses http-errors for proper error handling in Next.js API routes
+ */
+export function createPermissionError(
+  code: PermissionErrorCode,
+  statusCode: number = 403,
```

Comment:
I think this is great, could perhaps be used in some more places in the future 👍🏼